### PR TITLE
Normalize package paths to always contain a trailing slash

### PR DESCRIPTION
**Pull request by zickgraf**
`SetPackagePath` always appends a trailing "/" to the installation path of a package while `AddPackageInfos` did not before this PR. This could lead to the following error:

```gap
gap> SetPackagePath( "cap", GAPInfo.PackagesInfo.cap[1].InstallationPath );
gap> LoadPackage( "CAP", false );
true
gap> ExtendRootDirectories( [ "/tmp/some/dir" ] );
gap> SetPackagePath( "cap", GAPInfo.PackagesInfo.cap[1].InstallationPath );
Error, another version of package cap is already loaded at .../lib/package.gi:1752
brk> GAPInfo.PackagesLoaded.( pkgname )[1];
".../.gap/pkg/CAP_project/CAP/"
brk> pkgpath;
".../.gap/pkg/CAP_project/CAP"
```
i.e. an error is signaled because the installation path differs by a trailing slash.

This PR does two things to prevent this:
1. It normalizes the installation path in `AddPackageInfos` to also include a trailing slash.
2. It uses `Directory` in the comparison which triggers the error.

Both changes would fix the error on their own, but I think both are sensible in general:

1. Being consistent is always a good thing.
2. `pkgpath` is a user input, so normalizing it is a good idea anyway (as users might enter paths including "~" or with or without trailing slashes).

## Text for release notes

none

### CODE REVIEW

**Comment by ChrisJefferson** [comment]
This seems fine to me in principle, but I'll be honest I've never used `SetPackagePaths` or `ExtendRootDirectories`. Some tests for this, perhaps in `testinstall/package.tst` using the existing `mockpkg` might work?

**Comment by zickgraf** [comment]
> This seems fine to me in principle, but I'll be honest I've never used `SetPackagePaths` or `ExtendRootDirectories`.

Background: I use `SetPackagePath` to enforce loading packages from the user root dir in the CI, and our Julia package [CapAndHomalg.jl](https://github.com/homalg-project/CapAndHomalg.jl) uses `ExtendRootDirectories` to add its own root with our packages.

> Some tests for this, perhaps in `testinstall/package.tst` using the existing `mockpkg` might work?

Done!

File: tst/testinstall/package.tst [hunk 1]
```diff
@@ -640,5 +640,21 @@ false
 gap> IsPackageLoaded("mockpkg", ">=2.0");
 false
 
+# now add `tst` as a new root directory
```
[the remark this answers is not in the document]
Comment (zickgraf) [reply tst/testinstall/package.tst:643]: Two questions:

1. Can we rely on `IO_symlink` in the test? If not, I guess things might get messy.
2. What about Windows, would `IO_symlink` work there? Or can we somehow exclude this test on Windows?

[the remark this answers is not in the document]
Comment (ChrisJefferson) [reply tst/testinstall/package.tst:643]: In simple terms, you can assume everything unixy will work in Windows. I'll be sure to double check any test, but I just tried and IO_symlink works fine.

[the remark this answers is not in the document]
Comment (fingolfin) [reply tst/testinstall/package.tst:643]: I would not rely on `IO_symlink`, which may not always be available (if `IO` is not compiled). You could simply use `Exec("ln -sf FOO BAR")`, or if you want to be able to detect errors, then use `Process` (and if anyone wants a nicer alternative, comment on #5103 ???)

[the remark this answers is not in the document]
Comment (zickgraf) [reply tst/testinstall/package.tst]: I have pushed a solution using `Exec`. If an error occurs, `ValidatePackageInfo` right after the Exec will fail saying that the file is not readable, so I think there is no need for an additional error check .

File: tst/testinstall/package.tst [hunk 2]
```diff
@@ -418,8 +418,17 @@ false
...
+# load mockpkg via a symlink in a directory called `pkg`
+# so we can test ExtendRootDirectories below
+# first create a temporary directory for all of this
+gap> tmp_dir := DirectoryTemporary( );;
+
+# create a subdirectory `<tmp_dir>/pkg`
+gap> Exec( Concatenation( "mkdir -p ", Filename( tmp_dir, "/pkg" ) ) );
+
+# make `<tmp_dir>/pkg/mockpkg` a symlink to `tst/mockpkg`
+gap> Exec( Concatenation( "ln -sfn ", Filename( DirectoriesLibrary("tst/mockpkg")[1], "" )," ", Filename( tmp_dir, "pkg/mockpkg" ) ) );
```
Comment (fingolfin) [inline tst/testinstall/package.tst:430]:
```suggestion
gap> Exec( Concatenation( "ln -sfn ", Filename( DirectoriesLibrary("tst/mockpkg"), "" )," ", Filename( tmp_dir, "pkg/mockpkg" ) ) );
```
It should not be necessary to use `[1]` here.

**Review by fingolfin — APPROVED** [review]

File: tst/testinstall/package.tst (hunk 2 above)
Comment (zickgraf) [reply tst/testinstall/package.tst]: Done!

**Pull request merged**